power_grid_model_c/power_grid_model/include/power_grid_model/optimizer/base_optimizer.hpp

```diff
@@ -52,8 +52,8 @@ class BaseOptimizer {
 
     template <std::derived_from<BaseOptimizer> Optimizer, typename... Args>
         requires std::constructible_from<Optimizer, Args...>
-    static auto make_shared(Args... args) -> std::shared_ptr<BaseOptimizer> {
-        return std::static_pointer_cast<BaseOptimizer>(std::make_shared<Optimizer>(std::move(args)...));
+    static auto make_shared(Args&&... args) -> std::shared_ptr<BaseOptimizer> {
+        return std::static_pointer_cast<BaseOptimizer>(std::make_shared<Optimizer>(std::forward<Args>(args)...));
```

Comment:
@mgovers the old code is not correct. It unconditionally moves the object. If the input argument is a lvalue, that object will get consumed. This was the cause of address sanitizer warning.